src/coprocessor/dag/rpn_expr/impl_cast.rs

```diff
@@ -33,6 +34,63 @@ pub fn get_cast_fn_rpn_node(
         (EvalType::DateTime, EvalType::Real) => cast_time_as_real_fn_meta(),
         (EvalType::Duration, EvalType::Real) => cast_duration_as_real_fn_meta(),
         (EvalType::Json, EvalType::Real) => cast_json_as_real_fn_meta(),
+        (EvalType::Int, EvalType::Int) => {
```

Comment:
Could we use `match` instead?

Reply:
The `i` version will generate fewer ASM instructions than the `m1` version.
```asm
playground::m1:
	subq	$32, %rsp
	movb	%sil, %al
	movb	%dil, %cl
	andb	$1, %cl
	movb	%cl, 6(%rsp)
	andb	$1, %al
	movb	%al, 7(%rsp)
	movb	6(%rsp), %al
	movb	7(%rsp), %cl
	andb	$1, %al
	movb	%al, 24(%rsp)
	andb	$1, %cl
	movb	%cl, 25(%rsp)
	testb	$1, 24(%rsp)
	jne	.LBB59_5
	jmp	.LBB59_6
```
```asm
playground::i:
	subq	$24, %rsp
	movb	%sil, %al
	movb	%dil, %cl
	andb	$1, %cl
	movb	%cl, 6(%rsp)
	andb	$1, %al
	movb	%al, 7(%rsp)
	testb	$1, 6(%rsp)
	je	.LBB61_6
	testb	$1, 7(%rsp)
	je	.LBB61_5
	leaq	.L__unnamed_6(%rip), %rax
	movq	%rax, 8(%rsp)
	movq	$10, 16(%rsp)
```

Reply:
Anyway, I have changed to the `match` patter for code readability.

Reply:
Both styles look good to me..